### ISSUE TITLE
delaunay.find returns zero when it should not

### ISSUE THREAD

**Issue by boygirl** [d3/d3-delaunay#92]
I maintain a cloned and slimmed down version of `d3-delaunay`, and recently had the following edge case brought to my attention:

```
var scaled_data = [
  [60    ,17113.1],
  [106.5 ,17113.1],
  [153   ,17113.1],
  [199.5 ,17113.1],
  [246   ,17113.1],
  [292.5 ,17113.1],
  [339   ,17113.1],
  [385.5 ,17113.1]]

let delaunay = Delaunay.from(scaled_data)

delaunay.find(300, 17113.1)  // returns 0 but should return 5
```
 I find that it fails when added to your test suite as well. 

See https://github.com/FormidableLabs/delaunay-find/issues/9 for original report

**Comment by Fil** [comment]
From what I see, it comes from the poor precision of [area](https://github.com/d3/d3-delaunay/blob/master/src/delaunay.js#L16), which in this case return ~2e-10 (instead of 0 as the points are collinear). As a consequence, the test at https://github.com/d3/d3-delaunay/blob/master/src/delaunay.js#L53 is skipped, and the collinear "hack" doesn't kick in. Which means we have no neighbors for (0), and find stops there.

tldr: We need a better formula for area.

**Comment by boygirl** [comment]
Would it also work to alter the threshold for collinearity to be a bit higher than the current value of `1e-10`? Perhaps to match the jitter constant of `1e-8`? I don't understand where these values are coming from, so forgive me if this is an unhelpful suggestion.

**Comment by Fil** [comment]
We can fiddle and move thresholds around, but the risk is that, while it would solve the issue for this specific set of values, it might break for another. What we want is probably a larger set of test cases, a more precise area computation, or a different strategy.

**Comment by mourner** [comment]
Maybe try https://en.wikipedia.org/wiki/Kahan_summation_algorithm ?

**Comment by Fil** [comment]
just tried this in my notebook and the area is evaluated as 1.1641532182693481e-10 ; better than 2.3283064365386963e-10 but still not good enough.

A way to get to a very nice and round 0 is to substract hull[0]'s coordinates from each subsequent point… but isn't this cheating?

**Comment by Fil** [comment]
Ref. https://github.com/d3/d3-array/pull/35

**Comment by Fil** [comment]
In this case, we can tell the collinearity because the triangles returned is empty. See #93 for a solution that includes this, a kahan summation and a translation by -point[hull[0]].

**Comment by Fil** [comment]
I've updated #93 with much saner solution (faster, more accurate, and with less code).

**Comment by Fil** [comment]
fixed in https://github.com/d3/d3-delaunay/commit/c403f4bde57d68604b2cba79369f278bbd868362

thank you Lauren for the report
